### DISCUSSION TITLE
Installation of whipser on ubuntu automatically installs CUDA related libraries even though I do not have a GPU

### DISCUSSION THREAD

**Discussion by qaixerabbas** [openai/whisper#1514] [Q&A]
While installing whisper on EC2 (ubuntu) I can see the installation of CUDA related libraries. In the requirements.txt there isn't any CUDA related library. On my EC2 instance there is no GPU I am just using a general purpose medium instance. I do not want to install these additional CUDA related libraries. Any help in this regard shall be highly appreciated.

**Comment by glangford** [comment]
I don't use this platform combination, so this is a guess - the inclusion of CUDA "stuff" may be triggered by pytorch. Before whisper you could try installing pytorch from the cpu-only wheel, something like

`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu`

But if anyone else actually using Ubuntu and EC2 replies here, try their suggestion first 😊

**Reply by qaixerabbas** [reply]
Hey @glangford thank you for your suggestion. Actually, I think its not about EC2, its about Ubuntu I guess. Last time when I was dockerizing a whisper application, I found same installation issues there as well. For that scenario, I added few commands to uninstall the CUDA related libraries. However, that step was manual, requiring an additional docker command, and I prefer the libraries to not be installed at all.
